### ISSUE TITLE
3.28.1 doesn't recognize filesize of fresh uploaded files

### ISSUE THREAD

**Issue by IngoBelka** [nextcloud/android#12822]
### ⚠️ Before posting ⚠️

- [X] This is a **bug**, not a question or an enhancement.
- [X] I've [searched for similar issues](https://github.com/nextcloud/android/issues) and didn't find a duplicate.
- [X] I've written a clear and descriptive title for this issue, not just "Bug" or "Crash".
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Steps to reproduce

1. upload a file
2. open app version 3.28.1
3. search for uploaded file

### Expected behaviour

to see the right filesize and to be able ro download the file

### Actual behaviour

filesize 0B - no download possible 

### Android version

9 / 11

### Device brand and model

Huawei mediapad 5 / Xiaomi Mi 9t 

### Stock or custom OS?

Stock

### Nextcloud android app version

3.28.1

### Nextcloud server version

27.1.6

### Using a reverse proxy?

I don't know

### Additional information

There are no problems with version 3.28.

**Comment by dajusc** [comment]
Same Problem here with NC 28.0.4 and Android App 3.28.1. 
All files are affected, not only new ones. 
As soon as I try to download any file, the displayed size of this file changes to "0 B" and the file is not downloaded.
It worked fine with the previous Android App 3.28.0.

**Comment by brtwrst** [comment]
I have the same problem as described above. Downloading files or opening pdf files is not working. Exporting files (to the downloads folder) is still working.
There seem to be several issues with the 3.28.1.
I have downgraded to 3.28.0 for now and everything is back to normal.

**Comment by Discostu36** [comment]
> I have downgraded to 3.28.0 for now and everything is back to normal.

Uninstall, get apk from GitHub, reinstall and set up again? Or is there an easier way for downgrading?

**Comment by brtwrst** [comment]
I just use the Fdroid App Store, when the app is uninstalled and you visit the site for the nextcloud app, click the 3 dots top right and disable the updates, then go to the bottom and you can find the "available versions" menu where you can chose what version to install.

**Comment by igor-cali** [comment]
I think this [post](https://help.nextcloud.com/t/mobile-client-on-android-does-not-synchronize/177838/6) of mine in the community can be related to this bug. I paste my findings below for convenience.

> Trying to trobleshoot this myself, I note that the files is shown as syncronized in the Android client (automatically or if synced manually from the … file item menu) but:
> 
>     /storage/emulated/0/Android/media/com.nextcloud.client/[myuserfolder]/[sub-path]/[document.ext]: contains an old version with old filestamp
>     /storage/emulated/0/Android/media/com.nextcloud.client/tmp/[myuserfolder]/[sub-path]/[document.ext]: note the tmp in the path - contains the most updated version with updated filestamp.
> 
> Manually moving from the tmp folder to the proper one works it around until next sync. It looks like the sync process is not completed in the last step (from tmp to the proper folder.

**Comment by Pal-FC** [comment]
Same problem With the latest version of NEXTCLOUD(3.28.1), only with opening PDFs.
For other file types (jpg, md, etc..) no problem.

**Comment by Discostu36** [comment]
I don't think this is PDF specific, I have the same problem with my Keepass database file.

**Comment by Discostu36** [comment]
Thanks, seems to work. The only thing I noticed is that in the "activity" view the file size of every version is still shown as "0 B".

**Comment by IngoBelka** [comment]
Oh, that's a good news. So, I'm waiting for F-Droid delivering that new version. Thank you!

**Comment by reaper** [comment]
Great ! waiting for gplay release, thanks!

**Comment by monochromec** [comment]
Confirming that version 3.28.2 indeed fixes this. What was strange is that the logs generated by the app all reported a "200" status when downloading a file....

**Comment by Pal-FC** [comment]
> Confirming that version 3.28.2 indeed fixes this. What was strange is that the logs generated by the app all reported a "200" status when downloading a file....


I don't understand, 200 status is the http response from the server, it's not an error...

**Comment by ulrikrasmussen** [comment]
I can confirm that this is fixed if downloading and installing the 3.28.2 gplay apk on my phone. But it is a bit disconcerting that there is no PR fixing this, no post-mortem of what went wrong, and no mention in the release changelog.

This bug happened at the scariest possible time for me. Last night I had just used my phone to add a password to my Keepass database to a fairly important account with no other recovery options, and this morning the database showed as having 0 bytes. I had to pull the power from my router and start my laptop to backup the database stored there, then bring it online and luckily realize that the changes from last night had indeed been synced and that the file size of 0 bytes on my phone was bogus. Phew!

But honestly this has shaken by confidence in using Nextcloud to synchronize important data.

**Comment by Pal-FC** [comment]
I agree with you, there should have been at least a warning somewhere, but nothing.

**Comment by Pal-FC** [comment]
> I can confirm that this is fixed if downloading and installing the 3.28.2 gplay apk on my phone. But it is a bit disconcerting that there is no PR explicitly referencing this bug, no post-mortem of what went wrong, and no mention in the release changelog.
> 
> This bug happened at the scariest possible time for me. Last night I had just used my phone to add a password to my Keepass database to a fairly important account with no other recovery options, and this morning the database showed as having 0 bytes. I had to pull the power from my router and start my laptop to backup the database stored there, then bring it online and luckily realize that the changes from last night had indeed been synced and that the file size of 0 bytes on my phone was bogus. Phew!
> 
> But honestly this has shaken by confidence in using Nextcloud to synchronize important data.

I agree with you, there should have been at least a warning somewhere, but nothing.

**Comment by Szwendacz99** [comment]
For me, the version 3.28.2 from FDroid does not seem to fix the issue.

**Comment by igor-cali** [comment]
> Edit2: It actually looks like it just downloads those files that are in directory that i have opened at least once before, in the nc android app

I think this is the intended behaviour. It always worked for me in NC Android app in my devices: files/folder opened or downloaded are later synced (and marked with a green or red flag to show sync status).

**Comment by Szwendacz99** [comment]
> > Edit2: It actually looks like it just downloads those files that are in directory that i have opened at least once before, in the nc android app
> 
> I think this is the intended behaviour. It always worked for me in NC Android app in my devices: files/folder opened or downloaded are later synced (and marked with a green or red flag to show sync status).

It definitely was not working like that before. I do know that **files** once opened might be downloaded, but we are talking about whole directory sync.

**Comment by igor-cali** [comment]
> For me, the version 3.28.2 from FDroid does not seem to fix the issue.

Same for me: issue still observable with 3.28.2 from F-Droid. On Fairphone 3 device (Android 13).

**Comment by Whisprin** [comment]
@Szwendacz99 @igor-cali I had to manually delete the (empty) folder that contains the files. Afterwards syncing the file worked again with 3.28.2
The location where Nextcloud stores the files in the file system can be found under Settings > Data storage folder.

**Comment by monochromec** [comment]
> Confirming that version 3.28.2 indeed fixes this. What was strange is that the logs generated by the app all reported a "200" status when downloading a file....

While this works for a single file, it does't recursively synchronize a directory hierarchy as it was the case in previous versions.

**Comment by Notused1930** [comment]
Same problems here. Does anyone know when a solution to this will come

**Comment by mardukbp** [comment]
I downgraded to Version 3.23.0 and syncing a large directory tree full of PDFs works.

**Comment by p0ll3** [comment]
I got the same Problem on 3.29.0. The recommendation of whisprin to delete files worked for me, too. To be more precise. Look into the Settings where your Nextcloud files are cached. Something like: /storage/emulated/0/Android/media/com.nextcloud.client. In this Folder you will find a folder called nextcloud, open it. Then there you find a tmp folder. I deleted everything inside this tmp folder. After that, everything is working again.
